### PR TITLE
MAYA-112126: make sure parented object is added to selection list after it is moved.

### PR DESCRIPTION
**Pull request by HamedSabri-adsk**
**UFE Parent**
UFE_Parent

VS

**Maya Parent**
Maya_Parent

### CODE REVIEW

File: lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp [hunk 1]
```diff
@@ -28,9 +28,15 @@
 #include <pxr/usd/usd/stage.h>
 #include <pxr/usd/usdGeom/gprim.h>
 
+#include <maya/MGlobal.h>
+#include <maya/MItSelectionList.h>
+#include <maya/MSelectionList.h>
```
Comment (ppt-adsk) [inline lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp:33]: Do we need any of these?

File: lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp [hunk 2]
```diff
@@ -28,9 +28,15 @@
...
+#include <ufe/globalSelection.h>
 #include <ufe/log.h>
+#include <ufe/observableSelection.h>
 #include <ufe/scene.h>
 #include <ufe/sceneNotification.h>
+#include <ufe/selection.h>
```
Comment (ppt-adsk) [inline lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp:39]: Don't need this one, the UFE global selection is an ObservableSelection.

File: test/lib/ufe/testParentCmd.py [hunk 3]
```diff
@@ -1039,6 +1039,41 @@ def testParentingToGPrim(self):
...
+        globalSelection = ufe.GlobalSelection.get()
+        self.assertEqual(len(globalSelection), 0)
+
+        # parent cube to cylinder in relative mode, using UFE path strings.
+        cmds.parent("|mayaUsdProxy1|mayaUsdProxyShape1,/cubeXform",
+                    "|mayaUsdProxy1|mayaUsdProxyShape1,/cylinderXform",
+                    relative=True)
+
+        # parented object should be in the selection list now
+        self.assertEqual(len(globalSelection), 1)
```
Comment (ppt-adsk) [inline test/lib/ufe/testParentCmd.py:1068]: Would be stronger to check what's actually in the global selection, e.g.
cubePath = ufe.Path([shapeSegment, usdUtils.createUfePathSegment("/cylinderXform/cubeXform")])
cubeItem = ufe.Hierarchy.createItem(cubePath)
self.assertEqual(globalSelection.front(), cubeItem)

File: test/lib/ufe/testParentCmd.py [hunk 4]
```diff
@@ -1039,6 +1039,41 @@ def testParentingToGPrim(self):
...
+
+        cmds.undo()
+
+        self.assertEqual(len(globalSelection), 1)
```
Comment (ppt-adsk) [inline test/lib/ufe/testParentCmd.py:1072]: Same as line 1068, and same comment for line 1076.

**Review by ppt-adsk — APPROVED** [review]
Looks good.  Just a minor improvement to the test and we're good to go.  Thanks for the fix!

File: lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp (hunk 1 above)
Comment (HamedSabri-adsk) [reply lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp]: @ppt-adsk Ah sorry... I was experimenting with something and forgot to remove them. 

Fixed: cc3a604fa3d934e585d899813257eecd45702a8d

File: test/lib/ufe/testParentCmd.py (hunk 3 above)
Comment (HamedSabri-adsk) [reply test/lib/ufe/testParentCmd.py]: Good point. Thanks.

cc3a604fa3d934e585d899813257eecd45702a8d

**Review by ppt-adsk — APPROVED** [review]

**Comment by HamedSabri-adsk** [comment]
Setting this PR as ready-to-merge. Somehow this VM machine wasn't able to checkout gulark? Not really sure what is happening there. 

https://ecg-jenkins.ecs.ads.autodesk.com/job/ecg-mayausd-branch-preflight-2022-python3-windows/520/console

> CMake Error at gulark-subbuild/gulark-populate-prefix/tmp/gulark-populate-gitclone.cmake:75 (message):
>  Failed to checkout tag: '4e21ab305794f5309a1454b4ae82ab9a0f5e0d25'

**Pull request merged**